File: Platforms/Bioma/BiomaSurfaceSWATSoilSWATC/src/java/BiomaSurfaceSWATSoilSWATC/SurfaceSWATSoilSWATCExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SurfaceSWATSoilSWATCExogenous
{
    private Double AirTemperatureMinimum;
    private Double GlobalSolarRadiation;
    private Double WaterEquivalentOfSnowPack;
    private Double AirTemperatureMaximum;
    private Double Albedo;
    private Double AirTemperatureAnnualAverage;
    
    public SurfaceSWATSoilSWATCExogenous() { }
    
    public SurfaceSWATSoilSWATCExogenous(SurfaceSWATSoilSWATCExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.AirTemperatureMinimum = toCopy.getAirTemperatureMinimum();
            this.GlobalSolarRadiation = toCopy.getGlobalSolarRadiation();
            this.WaterEquivalentOfSnowPack = toCopy.getWaterEquivalentOfSnowPack();
            this.AirTemperatureMaximum = toCopy.getAirTemperatureMaximum();
            this.Albedo = toCopy.getAlbedo();
            this.AirTemperatureAnnualAverage = toCopy.getAirTemperatureAnnualAverage();
        }
    }
    public Double getAirTemperatureMinimum()
    { return AirTemperatureMinimum; }

    public void setAirTemperatureMinimum(Double _AirTemperatureMinimum)
    { this.AirTemperatureMinimum= _AirTemperatureMinimum; } 
    
    public Double getGlobalSolarRadiation()
    { return GlobalSolarRadiation; }

    public void setGlobalSolarRadiation(Double _GlobalSolarRadiation)
    { this.GlobalSolarRadiation= _GlobalSolarRadiation; } 
    
    public Double getWaterEquivalentOfSnowPack()
    { return WaterEquivalentOfSnowPack; }

    public void setWaterEquivalentOfSnowPack(Double _WaterEquivalentOfSnowPack)
    { this.WaterEquivalentOfSnowPack= _WaterEquivalentOfSnowPack; } 
    
    public Double getAirTemperatureMaximum()
    { return AirTemperatureMaximum; }

    public void setAirTemperatureMaximum(Double _AirTemperatureMaximum)
    { this.AirTemperatureMaximum= _AirTemperatureMaximum; } 
    
    public Double getAlbedo()
    { return Albedo; }

    public void setAlbedo(Double _Albedo)
    { this.Albedo= _Albedo; } 
    
    public Double getAirTemperatureAnnualAverage()
    { return AirTemperatureAnnualAverage; }

    public void setAirTemperatureAnnualAverage(Double _AirTemperatureAnnualAverage)
    { this.AirTemperatureAnnualAverage= _AirTemperatureAnnualAverage; } 
    
}

File: Models/BiomaSurfaceSWATSoilSWATC/src/cs/BiomaSurfaceSWATSoilSWATC/SurfaceSWATSoilSWATCExogenous.cs
using System;
using System.Collections.Generic;

public class SurfaceSWATSoilSWATCExogenous 
{
    private double _AirTemperatureMaximum;
    private double _AirTemperatureMinimum;
    private double _GlobalSolarRadiation;
    private double _WaterEquivalentOfSnowPack;
    private double _Albedo;
    private double[] _VolumetricWaterContent;
    
        public SurfaceSWATSoilSWATCExogenous() { }
    
    
    public SurfaceSWATSoilSWATCExogenous(SurfaceSWATSoilSWATCExogenous toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    AirTemperatureMaximum = toCopy.AirTemperatureMaximum;
    AirTemperatureMinimum = toCopy.AirTemperatureMinimum;
    GlobalSolarRadiation = toCopy.GlobalSolarRadiation;
    WaterEquivalentOfSnowPack = toCopy.WaterEquivalentOfSnowPack;
    Albedo = toCopy.Albedo;
    VolumetricWaterContent = new double[toCopy.VolumetricWaterContent.Length];
            for (int i = 0; i < toCopy.VolumetricWaterContent.Length; i++)
            { VolumetricWaterContent[i] = toCopy.VolumetricWaterContent[i]; }
    
    }
    }
    public double AirTemperatureMaximum
        {
            get { return this._AirTemperatureMaximum; }
            set { this._AirTemperatureMaximum= value; } 
        }
    public double AirTemperatureMinimum
        {
            get { return this._AirTemperatureMinimum; }
            set { this._AirTemperatureMinimum= value; } 
        }
    public double GlobalSolarRadiation
        {
            get { return this._GlobalSolarRadiation; }
            set { this._GlobalSolarRadiation= value; } 
        }
    public double WaterEquivalentOfSnowPack
        {
            get { return this._WaterEquivalentOfSnowPack; }
            set { this._WaterEquivalentOfSnowPack= value; } 
        }
    public double Albedo
        {
            get { return this._Albedo; }
            set { this._Albedo= value; } 
        }
    public double[] VolumetricWaterContent
        {
            get { return this._VolumetricWaterContent; }
            set { this._VolumetricWaterContent= value; } 
        }
}